### ISSUE TITLE
AssetId always converted to Multilocation or errors with numbers other than 0

### ISSUE THREAD

**Issue by Robiquet** [polkadot-js/api#5846]
Bug report

At Zeitgeist we use `pallet-asset-tx-payment` to pay fees in other currencies. We use it like this:
```        
const keyring = new Keyring({ type: "sr25519" });
const alice = keyring.addFromUri("//Alice", { name: "Alice default" });
const signedTx = await tx.signAsync(alice, {assetId: 0});
```

However now that we are using the latest version `assetId` is being forced into a `Multilocation` or in other cases it just errors. 

Here are some examples:

```
const signedTx = await tx.signAsync(alice, {assetId: 0});
console.log(signTx.assetId) // {"parents": "0","interior": "Here"}
```

```
const signedTx = await tx.signAsync(alice, {assetId: 1});
console.log(signTx.assetId) // "Error: Struct: failed on assetId: Option<MultiLocation>:: Struct: Cannot decode value 1 (typeof number), expected an input object, map or array"
```
Moving forward we would like to use arbitrary values like this but run into similar issues:
```
const signedTx = await tx.signAsync(alice, {assetId: { ForeignAsset: 0 }});
console.log(signTx.assetId) // {"parents": "0","interior": "Here"}
```

```
const signedTx = await tx.signAsync(alice, {assetId: { ForeignAsset: 1 }});
console.log(signTx.assetId) // {"parents": "0","interior": "Here"}
```

Seems related to the changes made in this pr: https://github.com/polkadot-js/api/pull/5752

* **Please tell us about your environment:**

  - Version: 10.12.6
  - Environment:

    - [x] Node.js
    - [x] Browser
    - [ ] Other (limited support for other environments)

  - Language:

    - [ ] JavaScript
    - [x] TypeScript (include tsc --version)
    - [ ] Other

**Comment by sea212** [comment]
Is there a way to use anything else than `MultiLocation` as `AssetId`?

One pallet used to allow transaction fee payment with assets is a standard Substrate frame pallet [`pallet-asset-tx-payment`](https://github.com/paritytech/polkadot-sdk/tree/832570545b0311f533499ead57d764a2bc04145c/substrate/frame/transaction-payment/asset-tx-payment). In the transaction validation function, it uses an `AssetId` that [can be specified by the developer](https://github.com/paritytech/polkadot-sdk/blob/832570545b0311f533499ead57d764a2bc04145c/substrate/frame/transaction-payment/asset-tx-payment/src/lib.rs#L152). It seems wrong to me to force the asset id type within the sdk when it is configurable in the pallet that processes the asset id on the blockchain side.

**Comment by TarikGul** [comment]
> Is there a way to use anything else than MultiLocation as AssetId?

Yes, you should be able to do this by following what I wrote in the above:

> If I am not mistaken you can add any [typesBundle or typesSpec](https://github.com/polkadot-js/api/blob/master/packages/types/src/types/registry.ts#L97) fields to override certain types when instantiating the ApiPromise. And you would set TAssetConversion to Option<AssetId> as seen here in the PR you [referenced](https://github.com/polkadot-js/api/pull/5752/files#diff-ed559e84e01ac9b0166c3e3cba78445a16093a155349ca7b889400331fa2c26cR53).

**Comment by Robiquet** [comment]
Thanks @TarikGul we managed to get it working with [this change](https://github.com/zeitgeistpm/sdk-next/pull/50/files) and signing with a keyring pair, see transaction [here](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fbsr.zeitgeist.pm#/explorer/query/0xf9d3071a70c01306d53a76152e1565765134af84ea573c66ee0edc7baee056f5). However when we try to sign with an extension we get the following error: 

![image](https://github.com/polkadot-js/api/assets/4950844/306531b0-b570-41be-ba31-80b0a82323bc)

I'm guessing this means that we need to add the overrides to this repo in a similar fashion to the asset hubs, what do you think?

**Comment by Robiquet** [comment]
Hi @TarikGul would you be able to take a look at this and let us know the best course of action?

**Comment by TarikGul** [comment]
> I'm guessing this means that we need to add the overrides to this repo in a similar fashion to the asset hubs, what do you think?

I am not completely sure to be honest, what version of the extension are you using?

**Comment by Robiquet** [comment]
I'm using version 0.44.1. I think this is the issue because it looks very similar to [this](https://github.com/polkadot-js/extension/issues/1314). It looks like the reason for it was that [these changes](https://github.com/polkadot-js/api/pull/5752/files) to override the types had not yet been deployed. So it seems like we will need to add similar overrides for Zeitgeist

**Comment by gagi13** [comment]
Hi guys, happy this thread exists 😁 

We are developing the UI for [ACP](https://crates.io/crates/pallet-asset-conversion) and ran into the same issue. So when we tried to sign the transaction with a non-DOT token as gas fee token via extension, it broke and we got the same error as above. Any updates on resolving this or any workaround you found?  @Robiquet 

FYI @fmol2y

**Comment by Robiquet** [comment]
@gagi13 I not too sure about that as the correct overrides should already be there for the asset hubs. I would try different wallets and versions of polkadot js. It seems like there hasn't been a release published for polkadot js extension for a while now.

**Comment by TarikGul** [comment]
We are very close to releasing a new version of the extension.

Also it might be worth checking to see if this works against the most recent version of the extension in github, instructions on how to run it in the [README](https://github.com/polkadot-js/extension?tab=readme-ov-file#development-version)

**Comment by TarikGul** [comment]
New versions of the extension have both been released for a while now: @Robiquet is this still an issue?

**Comment by Robiquet** [comment]
> New versions of the extension have both been released for a while now: @Robiquet is this still an issue?

Yeah the new release of the extension has made things worse for us, now neither approach works. Seems like overrides within polkadot js api are needed, the same sort of changes that were done to support the asset hubs

**Comment by Chralt98** [comment]
We will just adapt our chain to use `MultiLocation` instead. Is there any problem with using `MultiLocation` version 3? I am asking, because this is what our current runtime supports.

**Comment by Chralt98** [comment]
I experience difficulties when testing: https://github.com/polkadot-js/api/issues/6074

**Comment by Chralt98** [comment]
Ok, even the `MultiLocationV3` is working without any overrides to the current default of v4. This issue has now been resolved, thanks!
